### ISSUE TITLE
Querying locations of objects...

### ISSUE THREAD

**Issue by dicroce** [nyanp/tiny-cnn#68]
Is it possible to locate specific feature in an image and get their locations from tiny-cnn?

If I train a NN to detect say sail boats... and then I get a positive result... is their a way to determine what part of the image contributed to the positive result?

**Comment by micahpearlman** [comment]
For our project we used a sliding window to get location.  The nature of CNN is that you lose location information.

**Comment by dicroce** [comment]
Really? What if I want to detect multiple objects in the same image? That is potentially a lot of sliding windows! Is this the normal way this is done?

**Comment by nyanp** [comment]
Sliding window is one of the normal way to get object location. CNN for each sliding window is very heavy, so extract region proposals using lightweight detector (Haar Cascades in face detection, Bing detector for generic detection task, for example) before CNN is widely used. R-CNN is one of the most popular method.

Girshick, R., Donahue, J., Darrell, T.,&Malik, J. [Rich feature hierarchies for accurate object detection and semantic segmentation](http://arxiv.org/pdf/1311.2524v5.pdf)

**Comment by stereomatchingkiss** [comment]
You could check the paper "[What makes for effective detection proposals](http://arxiv.org/abs/1502.05082)"?

In short, selective search and edgeboxes give best results, but edgeboxes is much faster than selective search. dlib ad contribute module of opencv, both of them provide selective search, but they still lack edge boxes.

**Comment by edgarriba** [comment]
@dicroce If you come up with an implementation of this paper don't hesitate to give feedback.

**Comment by Randl** [comment]
@dicroce Have  you managed to solve the problem?

**Comment by dicroce** [comment]
Nope. And I'd still like to know how to do this... I've just been
sidetracked for a while.

**Comment by Randl** [comment]
@dicroce Have you tried [SSD](https://github.com/weiliu89/caffe/tree/ssd)?

**Comment by pliptor** [comment]
@dicroce  I'm not entirely sure if this project has anything relevant for your objectives but it appears the author is pretty happy with the results using tiny-dnn.
[OpenIST](https://github.com/pliptor/OpenIST)
